Saule/Serialization/ResourceSerializer.cs

```diff
@@ -96,10 +97,12 @@ private ResourceGraphPathSet IncludedGraphPathsFromContext(IncludingContext cont
 
         private JToken CreateTopLevelLinks(int count)
         {
-            var result = new JObject
+            var result = new JObject();
+
+            if (_resource.WithLinks.HasFlag(LinkTypes.Self & LinkTypes.All))
```

Comment:
I think you can just do `HasFlag(LinkTypes.Self)`, since `All` includes that. Now this is only added when the value is `LinkTypes.All`.